backtracking/minimax.py

```diff
@@ -35,36 +50,45 @@ def minimax(
     12
     """
 
+    # Check for invalid inputs
     if depth < 0:
         raise ValueError("Depth cannot be less than 0")
-
     if len(scores) == 0:
         raise ValueError("Scores cannot be empty")
 
+    # Base case: If the current depth equals the height of the tree,
+    # return the score of the current node.
     if depth == height:
         return scores[node_index]
 
+    # If it's the maximizer's turn, choose the maximum score
+    # between the two possible moves.
     if is_max:
         return max(
             minimax(depth + 1, node_index * 2, False, scores, height),
             minimax(depth + 1, node_index * 2 + 1, False, scores, height),
         )
 
+    # If it's the minimizer's turn, choose the minimum score
+    # between the two possible moves.
     return min(
         minimax(depth + 1, node_index * 2, True, scores, height),
         minimax(depth + 1, node_index * 2 + 1, True, scores, height),
     )
 
 
 def main() -> None:
+    # Sample scores and height calculation
     scores = [90, 23, 6, 33, 21, 65, 123, 34423]
     height = math.log(len(scores), 2)
+
+    # Calculate and print the optimal value using the minimax algorithm
     print("Optimal value : ", end="")
     print(minimax(0, 0, True, scores, height))
 
 
 if __name__ == "__main__":
     import doctest
 
-    doctest.testmod()
-    main()
+    doctest.testmod()  # Run doctests to ensure the function behaves as expected
+    main()  # Run the main function to demonstrate the minimax algorithm in action
```

Comment:
```suggestion
    doctest.testmod()
    main()
```
Likewise, I feel that this is too simple to need any comments